### ISSUE TITLE
Cluster monitoring does not work in cluster with restricted psp as default

### ISSUE THREAD

**Issue by axeal** [rancher/rancher#20229]
**What kind of request is this (question/bug/enhancement/feature request):**

Bug

**Steps to reproduce (least amount of steps as possible):**

- Create a custom cluster with pod security policy enabled and the default set to restricted
- After the cluster has been created, enable cluster monitoring

**Result:**

`grafana-proxy` container in the `grafana-cluster-monitoring` deployment errors with:
```
2019/05/14 11:18:18 [warn] 1#1: duplicate MIME type "text/html" in /nginx/nginx.conf:60
nginx: [warn] duplicate MIME type "text/html" in /nginx/nginx.conf:60
2019/05/14 11:18:18 [emerg] 1#1: chown("/tmp/temp_dir", 100) failed (1: Operation not permitted)
nginx: [emerg] chown("/tmp/temp_dir", 100) failed (1: Operation not permitted)
```

`grafana-init-json-modify` container in the `grafana-cluster-monitoring` deployment errors with:
```
/run.sh: line 14: can't create /host/grafana/plugin.json: Permission denied
```

`prometheus-proxy` in the `prometheus-cluster-monitoring` stateful set errors with:
```
2019/05/14 11:20:00 [warn] 1#1: duplicate MIME type "text/html" in /var/run/nginx.conf:50
nginx: [warn] duplicate MIME type "text/html" in /var/run/nginx.conf:50
2019/05/14 11:20:00 [emerg] 1#1: chown("/tmp/temp_dir", 100) failed (1: Operation not permitted)
nginx: [emerg] chown("/tmp/temp_dir", 100) failed (1: Operation not permitted)
```

**Other details that may be helpful:**

Root cause similar to https://github.com/rancher/rancher/issues/16119, i.e. lack of unrestricted psp grant to service accounts in all system project namespaces

Issue can be worked around by applying the following to the cluster:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: default-psp-role
  namespace: cattle-prometheus
rules:
- apiGroups:
  - extensions
  resourceNames:
  - default-psp
  resources:
  - podsecuritypolicies
  verbs:
  - use

---

apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: default-psp-rolebinding
  namespace: cattle-prometheus
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: default-psp-role
subjects:
- kind: Group
  name: system:serviceaccounts:cattle-prometheus
  apiGroup: rbac.authorization.k8s.io
```

**Cluster information**
Rancher v2.2.2

**Comment by janeczku** [comment]
Note that this issue is not limited to cluster monitoring. It equally occurs when trying to enable project monitoring in a project configured with a restricted PSP.

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher:master-head f0392794b

Steps: the same as the comment for reproduction

Results:
- both the cluster and project monitoring are active and functional 
- do not see the errors mentioned in the issue anymore
